devdoc/channel_2.md

```diff
@@ -0,0 +1,72 @@
+# channel_2 requirements:
+
+## Overview
+
+The current implementation of `channel` has certain pitfalls. `channel_2` aims to address these pitfalls by making the following changes:
+
+1. Introduce a `channel_2_reset` API: This API will allow the user to reset the `channel_2` to a clean state. It will cause all pending operations to be abandoned.
```

Comment:
as written, `channel_2_reset` is a synchronous API, and I'm assuming that it can be called in SM_OPENED.  How does this API interact with other API calls?  Does it use `sm_exec_begin` or `sm_barrier_begin`? My worry here is that I don't know what happens to other operations while this reset operation is taking place?  If I call `channel_2_push` around the same time a different thread calls `c hannel_2_reset`, what happens?  I assume three are some cases where my push will be accepted, then the reset will start, and my push will be cancelled.  But what if the reset starts before the push is accepted?  What happens then?  Does my thread block until the reset is complete?  This could be a while. Or, does reset make a copy of the current "queue" so it can cancel the existing operations while the channel is accepting new ones?

This is all assuming that `c hannel_2_reset` doesn't return until after all of the `CHANNEL__2_PUSH_CALLBACK` and `CHANNEL_2_PULL_CALLBACK` calls have been called.  Is this guaranteed?  I assume so. OPtherwise this function only starts the reset process without letting the caller know when the reset process is complete.